bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -415,9 +415,8 @@ private Object readResolve() {
     }
 
     private static byte[] parseHexString(final String s) {
-        if (!isValid(s)) {
-            throw new IllegalArgumentException("invalid hexadecimal representation of an ObjectId: [" + s + "]");
-        }
+        notNull("hexString", s);
```

Comment:
> should we perhaps add a test case to verify the behavior of passing a 24-character string that does not resolve to a hexadecimal

I did add that,  on line 150 of the test file.

> it would be more helpful for the above to return invalid hexadecimal: [MY_BAD_STRING_THAT_IS_24] than invalid hexadecimal: M

I'm indifferent about this, since I don't think it's common and probably can be worked out from context.  But if you'd like to change it, I'll accept it.